tdd_project.rb

```
@@ -0,0 +1,18 @@
+class Solver
+  def factorial(num)
+    return nil unless num.is_a? Integer
+    return 1 if num <= 0
+
+    num * factorial(num - 1)
+  end
```

- [ ] You've done well by writing tests and implementing this method. But, one small part of the requirements for this particular task has not been met.

  Let's take another look at the requirement 👇🏼:
  
  > - Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
  
  From the above requirement, the following edge case has not been covered in both the tests and the method implementation itself 👇🏼: 
  
  `This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.`
  
  Please make sure that the method only accepts `0` and positive numbers. Otherwise, ensure that the method raises an exception. This should also be covered with tests.
  
  Currently, the function returns `1` for negative numbers(`num < 0`), which shouldn't be true.